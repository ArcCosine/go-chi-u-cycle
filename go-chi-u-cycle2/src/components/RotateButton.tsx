import { useState } from "react";

function RotateButton() {
    const [counter, setCounter] = useState(0);

    const countUp = () => {
        setCounter(counter + 1);
    };

    const countDown = () => {
        setCounter(counter - 1);
    };

    //useEffect(() => {
    //    window._gochiCounter = counter;
    //}, []);

    return (
        <div>
            <button type="button" onClick={countUp}>回転</button>
            <button type="button" onClick={countDown}>逆回転</button>
        </div>
    );
}
export default RotateButton;
