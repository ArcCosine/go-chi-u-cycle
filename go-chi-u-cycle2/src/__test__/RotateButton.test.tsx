import { describe, expect, test, afterEach } from "vitest";
import RotateButton from "../components/RotateButton";
import { render, screen, cleanup } from "@testing-library/react";

//afterEach(cleanup);

describe("Rotate button test", () => {
    test("button render test", () => {
        const { container } = render(<RotateButton />);
        expect(container).toMatchSnapshot();
    });

    test("check button nums", () => {
        render(<RotateButton />);
        expect(screen.getAllByRole("button")[0]).toBeTruthy();
        expect(screen.getAllByRole("button")[1]).toBeTruthy();
    });

    test("initial count", () => {
        const { container } = render(<RotateButton />);
        expect(screen.getByTestId("countup").textContent).toBe('0');
    });

    test("count up number", () => {
        const { container } = render(<RotateButton />);
        const buttonZero = screen.getAllByRole("button")[0];
        buttonZero.click();
        expect(screen.getByTestId("countup").textContent).toBe("1");
    });

    test("count down number", () => {
        const { container } = render(<RotateButton />);
        const buttonOne = screen.getAllByRole("button")[1];
        buttonOne.click();
        expect(screen.getByTestId("countup").textContent).toBe("-1");
    });
});
