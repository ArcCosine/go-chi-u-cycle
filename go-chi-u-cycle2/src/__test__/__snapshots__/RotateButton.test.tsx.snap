// Vitest Snapshot v1

exports[`Rotate button test > button render test 1`] = `
<div>
  <div>
    <div>
      <button
        type="button"
      >
        回転
      </button>
      <button
        type="button"
      >
        逆回転
      </button>
    </div>
    <div>
      <span
        id="counter"
      >
        0
      </span>
      回 回転させました。
    </div>
  </div>
</div>
`;
